pkg/didcomm/protocol/didexchange/service.go

```diff
@@ -12,24 +12,46 @@ import (
 	"errors"
 	"fmt"
 
+	"github.com/google/uuid"
+
+	"github.com/hyperledger/aries-framework-go/pkg/common/log"
 	"github.com/hyperledger/aries-framework-go/pkg/common/metadata"
 	"github.com/hyperledger/aries-framework-go/pkg/didcomm/dispatcher"
+	"github.com/hyperledger/aries-framework-go/pkg/didcomm/event"
 	"github.com/hyperledger/aries-framework-go/pkg/didcomm/protocol/decorator"
 	"github.com/hyperledger/aries-framework-go/pkg/didcomm/transport"
 	"github.com/hyperledger/aries-framework-go/pkg/storage"
 )
 
+// TODO https://github.com/hyperledger/aries-framework-go/issues/104
+var logger = log.New("aries-framework/didexchange")
+
+type didCommChMessage struct {
+	ID              string
+	DIDCommCallback event.DIDCommCallback
+}
+
 const (
 	// DIDExchange did exchange protocol
 	DIDExchange = "didexchange"
 	// DIDExchangeSpec defines the did-exchange spec
-	DIDExchangeSpec    = metadata.AriesCommunityDID + ";spec/didexchange/1.0/"
-	connectionInvite   = DIDExchangeSpec + "invitation"
-	connectionRequest  = DIDExchangeSpec + "request"
-	connectionResponse = DIDExchangeSpec + "response"
-	connectionAck      = DIDExchangeSpec + "ack"
+	DIDExchangeSpec = metadata.AriesCommunityDID + ";spec/didexchange/1.0/"
+	// ConnectionInvite defines the did-exchange invite message type.
+	ConnectionInvite = DIDExchangeSpec + "invitation"
+	// ConnectionRequest defines the did-exchange request message type.
+	ConnectionRequest = DIDExchangeSpec + "request"
+	// ConnectionResponse defines the did-exchange response message type.
+	ConnectionResponse = DIDExchangeSpec + "response"
+	// ConnectionAck defines the did-exchange ack message type.
+	ConnectionAck = DIDExchangeSpec + "ack"
 )
 
+type message struct {
```

Comment:
It would be good to have a comment on how message is being used.

Reply:
[Updated](https://github.com/hyperledger/aries-framework-go/compare/6ca38f34c66c2017dee287d5ecd2734ca169e7b7..5839b0f509b0932a67abd562fb6a05dc181f2e08#diff-f761c91e753eba1c42a68076df06ce12R50)